File: api/src/main/java/com/pauldaniv/template/client/DefaultTestClient.java
package com.pauldaniv.template.client;

import com.pauldaniv.template.api.TestOneService;
import com.pauldaniv.template.request.TestOne;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class DefaultTestClient implements TestClient {
    private final TestOneService testOneService;

    @Override
    public TestOne findByIsbn(String isbn) {
        return testOneService.getIt();
    }
}

File: api/src/main/kotlin/com/pauldaniv/template/client/DefaultTestClient.kt
package com.pauldaniv.template.client

import com.pauldaniv.template.api.TestOneService
import com.pauldaniv.template.request.TestOne
import org.springframework.stereotype.Component

@Component
class DefaultTestClient(private val testOneService: TestOneService) : TestClient {
  override fun findByIsbn(isbn: String): TestOne {
    return testOneService.getIt()
  }
}
